app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,88 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:created_at)
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id].to_i)
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_perems)
+
+    if @task.save
+      redirect_to task_path(@task.id)
+      return
+    else
+      render :new
```

Comment:
Just a suggestion to add the response code.

```suggestion
      render :new, status: :bad_request
```